shared/dto/widgets/base-widget-data.interface.ts

```diff
@@ -47,4 +47,5 @@ export type WidgetNavigationData = {
 
 export class BaseWidgetWithData extends BaseWidget {
   data: WidgetData;
+  responseRate: number;
```

Comment:
Being picky here: the `responseRate` is actually a transformation done exclusively in the client side, it shouldn't be shared as it won't be used by anything else.